### DISCUSSION TITLE
Custom controller when updating users from Admin Panel

### DISCUSSION THREAD

**Discussion by Bendrog** [strapi/strapi#7225] [Legacy Topics]
I want to custom the controller when someone update a User from the Admin Panel.

I have manage to custom this whenever someone updates a "classic" Content Type.
```cmd
[2020-07-28T08:07:42.525Z] debug PUT /content-manager/explorer/application::tickets.tickets/17 (1496 ms) 200
```

By overwriting the plugin `content-manager`.

In the path `/extensions/content-manager/controllers/ContentManager.js` Function update.

My problem is that for users it is a different route, because of `users-permisssions` plugin.

```cmd
[2020-07-28T08:09:55.559Z] debug PUT /content-manager/explorer/plugins::users-permissions.user/68 (1177 ms) 200
```

What folder should I overwrite to get this ?

**Comment by lauriejim** [comment]
Hello! Can yo please explain what you are trying to achieve?
This will help us to guide you in the right place.
Why you want to update that thing?
Thank for the details.

**Comment by Bendrog** [comment]
Hi,


The usecase is:

- Whenever someone switches the boolean `confirmed` to `true` in the Admin Panel, then we send an email to this person.

So I'd like to find that function linked to `PUT/content-manager/explorer/plugins::users-permissions.user/:id`

And add this `if` statement to send an email

**Reply by marefati110** [reply]
you can use lifecycle hooks
https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks

**Reply by lauriejim** [reply]
Thank you for sharing this link.
I suggest you to use the model's lifecycle functions to do that.

**Reply by Bendrog** [reply]
Thanks for your answers !

The problem with lifcycle is that I get if the boolean is true, but I do not know if it was true before.

It did the trick by using another boolean. 

- Default both booleans are `false`
- Switch Boolean A to `true` in Admin Panel
- Lifecycle checks if Boolean A is `true` and Boolean B is `false`. If so send email and Switch Boolean B to `true`


Knowing which functino to overwrite would be great to avoid this messy trick

**Reply by lauriejim** [reply]
You can easily fetch the data (so the current state) at the beginning of the lifecycle function.
Check if there is an update on this value.
If yes send the email 
If no, do nothing.
